include/ruby/io.h

```diff
@@ -69,158 +69,38 @@ typedef enum {
     RUBY_IO_PRIORITY = RB_WAITFD_PRI, /**< `IO::PRIORITY` */
 } rb_io_event_t;
 
-/**
- * IO  buffers.   This  is  an implementation  detail  of  ::rb_io_t::wbuf  and
- * ::rb_io_t::rbuf.  People don't manipulate it directly.
- */
-PACKED_STRUCT_UNALIGNED(struct rb_io_buffer_t {
-
-    /** Pointer to the underlying memory region, of at least `capa` bytes. */
-    char *ptr;                  /* off + len <= capa */
-
-    /** Offset inside of `ptr`. */
-    int off;
-
-    /** Length of the buffer. */
-    int len;
-
-    /** Designed capacity of the buffer. */
-    int capa;
-});
-
-/** @alias{rb_io_buffer_t} */
-typedef struct rb_io_buffer_t rb_io_buffer_t;
-
-/** Ruby's IO, metadata and buffers. */
-typedef struct rb_io_t {
-
-    /** The IO's Ruby level counterpart. */
-    VALUE self;
-
-    /** stdio ptr for read/write, if available. */
-    FILE *stdio_file;
-
-    /** file descriptor. */
-    int fd;
+struct rb_io_t;
```

Comment:
The `typedef` should be here too, to keep the source level compatibility.

Reply:
Ah yes, this was just my optimistic refactor :) I'll fix it, thanks!